### PR TITLE
Import recipes from github.com/anguslees/conda-*

### PR DESCRIPTION
**Pull request by anguslees**
Closes conda-forge/conda-forge.github.io#8

### CODE REVIEW

**Comment by anguslees** [comment]
I piled everything into one PR because it was easier for me.  Let me know if you would prefer one PR per recipe.

**Comment by msarahan** [comment]
Breaking things up would be nice.  You don't have to go all the way to one recipe per PR, but please limit it to related recipes in each PR.

Otherwise, this will take forever to review, and might get lost.

Thanks for adding your recipes!

**Comment by jakirkham** [comment]
So, some of these already exist so let's start by clearing those out first. Feel free to PR the feedstocks and add yourself as a maintainer on them.

* [amqp]( https://github.com/conda-forge/amqp-feedstock )
* [kombu]( https://github.com/conda-forge/kombu-feedstock ) 
* [vcversioner]( https://github.com/conda-forge/vcversioner-feedstock )

**Comment by jakirkham** [comment]
> Breaking things up would be nice. You don't have to go all the way to one recipe per PR, but please limit it to related recipes in each PR.

> Otherwise, this will take forever to review, and might get lost.

I think I encouraged him to dump them in elsewhere. Though its true. Review time is the biggest issue.

**Comment by jakirkham** [comment]
Also, get rid of `rabbitmq` and try to join forces on this PR ( https://github.com/conda-forge/staged-recipes/pull/471 ).

File: recipes/sqlalchemy-migrate/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    - setuptools
+    - pbr >=1.3
+
+  run:
+    - python
+    - "sqlalchemy >=0.7.8,!=0.9.5"
+    - six >=1.7.0
+    - decorator
+    - sqlparse
+    - tempita >-0.4
```
Comment (jakirkham) [inline recipes/sqlalchemy-migrate/meta.yaml]: Should it be `>=0.4`?

File: recipes/kombu/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    - kombu.tests.async
+    - kombu.tests.transport
+    - kombu.tests.transport.virtual
+    - kombu.tests.utils
+    - kombu.transport
+    #- kombu.transport.django
+    #- kombu.transport.django.management
+    #- kombu.transport.django.management.commands
+    #- kombu.transport.django.migrations
+    #- kombu.transport.sqlalchemy
```
Comment (jakirkham) [inline recipes/kombu/meta.yaml]: What's up with this?

**Comment by jakirkham** [comment]
Some general thoughts on the recipes.

* Licenses should be versioned (e.g. `Apache 2.0`, `BSD 3-Clause`, `LGPL 2.1`, etc.). Terse licenses are generally encouraged.
* If it is a simple python install script just add `setup.py install --single-version-externally-managed --record record.txt` to `build/script` and eliminate `build.sh` and `bld.bat`.
* Patches should be submitted upstream (they don't need to be accepted yet). Though do include a link to the upstream submission (e.g. PR for GitHub) and a brief summary in the `meta.yaml` about what the patch does and why we need it.

**Comment by jakirkham** [comment]
Thanks for the submission, @anguslees. I have tried to make some general notes about things that we can leave out to make this PR a little more review friendly. There were also a few things that stood out that I commented on. From a brief skim, these appear to be pretty healthy recipes.

**Comment by anguslees** [comment]
Sorry, I thought I had removed the duplicates, but must have messed up my conda search command somehow.  I'll try that process again now I have some candidates that _should_ be detected..

I'll break it up into smaller chunks too..

File: recipes/sqlalchemy-migrate/meta.yaml (hunk 1 above)
Comment (anguslees) [inline recipes/sqlalchemy-migrate/meta.yaml]: Yes, yes it should.  None of the tools ever balked at that - I guess they were doing "greater than a negative number"?

**Comment by jakirkham** [comment]
> Sorry, I thought I had removed the duplicates, but must have messed up my conda search command somehow.  I'll try that process again now I have some candidates that _should_ be detected..

No worries. I think those were the only ones. With the PR, it was named differently so not sure if you would have caught that.

> I'll break it up into smaller chunks too..

Sure, but feel free to leave this here as you do. That way you can kind of keep a check list of what still needs to move.

File: recipes/kombu/meta.yaml (hunk 2 above)
Comment (anguslees) [inline recipes/kombu/meta.yaml]: Erm ... I forget.
I'm guessing they're meant to be optional public entrypoints, but fail to import without the optional dependencies django and sqlalchemy installed.  See also `oslo.messaging/meta.yaml` where I had a similar problem.

In general, I wasn't sure what I should do with optional dependencies.  I can require them just for testing (which then means they need to be packaged, but I see django already is), or I can ignore them completely and not test them.  Is there a suggested approach for this?

**Comment by anguslees** [comment]
rabbitmq (#471) is rabbitmq-server, librabbitmq is a (python) client and not a duplicate, fwiw.

**Comment by jakirkham** [comment]
> rabbitmq (#471) is rabbitmq-server, librabbitmq is a (python) client and not a duplicate, fwiw.

Sorry, was doing my best to skim 160 files. 😜

**Comment by anguslees** [comment]
I think that addresses the general comments raised so far - I'll split it up and we can discuss what to do about optional dependencies and any other recipe-level comments there.

**Comment by jakirkham** [comment]
Well, I would add the BSD license need to state how many clauses. For instance, `BSD 2-Clause`.

**Comment by jakirkham** [comment]
Do these all work on Python 2.7, 3.4 and 3.5?

If some of the are Python 2 only, you should add `skip: true  # [py3k]` to the `build` section. Alternatively, if some are Python 3 only, change `py3k` to `py2k`.

**Comment by anguslees** [comment]
Hrm.  They all work with 2.7.  At least websockify 0.6.1 doesn't work with py3.  Newer versions of websockify *do* work with py3, but if I update it then I need to update lots of other things too (this is all old versions that I haven't updated in a while).

So: I feel like websockify 0.6 failing to build on py3 is maybe the desired result - some sort of test result dashboard *should* go red and indicate that we need to update it, and then after the update it will go green and we'll all be happy.  ... Or should I just mark it as py2-only, and then we remember out-of-band to remove that selector when I/we get around to uploading newer versions?

**Comment by jakirkham** [comment]
> Or should I just mark it as py2-only, and then we remember out-of-band to remove that selector when I/we get around to uploading newer versions?

👍

**Comment by jakirkham** [comment]
Just to point out something, these will all be separate repos. So issues of supporting previously unsupported platforms, Python versions, etc. can easily be tracked on those issue trackers.

We really don't want this repo (staged-recipes) to be the place where everything is perfected. This is more the place where things become good enough to be sent out to live on there own.

**Comment by jakirkham** [comment]
Any thoughts on breaking this up @anguslees? Could we grab maybe the first one in this PR to start and put it in a new PR to start?

**Comment by jakirkham** [comment]
Friendly nudge @anguslees.

**Comment by jakirkham** [comment]
So I have added PRs to add you as a maintainer on many of the recipes in here that are already feedstocks, @anguslees. If you want to generally tell me whether being added as maintainers to those feedstsocks are ok or not here, I can handle merging them all in without needing further response.

Also, I have added a PR ( https://github.com/anguslees/staged-recipes/pull/1 ) against your branch to clear out these recipes that already are feedstocks. In addition, I removed `linecache2` as that is already in PR ( https://github.com/conda-forge/staged-recipes/pull/887 ) and it seems basically ready to go. If you could please merge this when you have a chance, that should simplify the amount of work that needs to be done here.

Once we have handled the above tasks, we can take another look at this.

**Comment by anguslees** [comment]
@jakirkham: Thanks!  Yes, I'm fine with being added as a maintainer on those other feedstocks. https://github.com/anguslees/staged-recipes/pull/1 merged.

**Comment by jakirkham** [comment]
Great, thanks, @anguslees. Also, it appears I missed `python-mimeparse`. Please feel free to remove that one too.

**Comment by jakirkham** [comment]
So you should get an email inviting you to join conda-forge (probably in a few hours). Once you accept, you will be given permissions on all of those feedstocks (repos) where you are now listed as a maintainer. This will give you commit privileges on them and control over their CIs.

Please follow the GitHub workflow like with any other repo and fork them. If you need to make changes, please make them in your fork and PR them back to the main feedstock. This is very important to follow as every branch on the main repo autodeploys packages to conda-forge. So, you only want to make a change to them when you are sure you and others are happy with it.

Please let us know if you have any questions and welcome to conda-forge. 😄

**Comment by jakirkham** [comment]
Actually added the removal of `python-mimeparse` in this PR ( https://github.com/anguslees/staged-recipes/pull/2 ).

**Comment by patricksnape** [comment]
@jakirkham How are we feeling about pinning things these days? I just had a glance through the recipes on here and a lot of them have very specific valid version ranges.

**Comment by anguslees** [comment]
@patricksnape: I tried to pass through upstream's version requirements accurately.  A number of these packages come from openstack fwiw, so I'm guessing you're seeing the way openstack handles their dependencies.

**Comment by jakirkham** [comment]
Did you get an email to join conda-forge, @anguslees? If not please, let us know. Once the invitation is accepted, you should get permission on your feedstocks.

**Comment by anguslees** [comment]
@jakirkham: I cannot find such an invitation email, no.

**Comment by jakirkham** [comment]
We have `eventlet` now apparently. So that can be dropped. Proposing removal from this collection in PR ( https://github.com/anguslees/staged-recipes/pull/3 ). Please let me know if you are ok being a maintainer of it, @anguslees.

**Comment by jakirkham** [comment]
Some more things have been added in the interim or are ready to go in other PRs. So have proposed they get removed in PR ( https://github.com/anguslees/staged-recipes/pull/4 ).

**Comment by jakirkham** [comment]
Some more things have been added in the interim or are ready to go in other PRs. So have proposed they get removed in PR ( https://github.com/anguslees/staged-recipes/pull/4 ). Please merge when you have a chance, @anguslees.

**Comment by jakirkham** [comment]
Also added PRs ( https://github.com/conda-forge/fixtures-feedstock/pull/1 ) and ( https://github.com/conda-forge/testtools-feedstock/pull/1 ) to add you as a maintainer on those feedstocks, @anguslees. Please let me know if that is ok and they can be merged.

**Comment by jakirkham** [comment]
Toggling to re-run CIs.

Note to self: The linter correctly handled this case and did not raise a merge conflict thanks to PR ( https://github.com/conda-forge/conda-forge-webservices/pull/68 ).

File: recipes/os-brick/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  md5: 5bc611df7e5af5194b5d6fc1ce92a743
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record=record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
```
Comment (jakirkham) [inline recipes/os-brick/meta.yaml]: Needs `pbr` in `build` [apparently]( https://travis-ci.org/conda-forge/staged-recipes/builds/161214081#L518 ).

**Comment by jakirkham** [comment]
Doing a little more clean up in PR ( https://github.com/anguslees/staged-recipes/pull/5 ). Also would you mind enabling the [Allow edits from maintainers]( https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/ ) feature, @anguslees?

**Comment by anguslees** [comment]
Maintainer edits allowed, thanks for the reminder.

**Comment by jakirkham** [comment]
Thanks for doing that. Went ahead and merged that cleanup PR into here.

**Comment by jakirkham** [comment]
Toggling for CIs

**Comment by jakirkham** [comment]
Toggling for CIs.

**Comment by jakirkham** [comment]
@conda-forge-admin, please relint.

Trying to get an accurate idea of which of these are still needed vs. those that are already packaged.

**Pull request closed without merging**